includes/class-apple-news.php

```diff
@@ -53,4 +61,29 @@ public static function get_filename( $path ) {
 		// Remove any spaces and return the filename
 		return str_replace( ' ', '', $filename );
 	}
+
+	/**
+	 * Attempt to migrate settings from an older version of this plugin
+	 *
+	 * @param Settings $settings
+	 */
+	public function migrate_settings( $settings ) {
+		$migrated_settings = array();
+
+		// For each potential value, see if the WordPress option exists.
+		// If so, migrate its value into the new array format.
+		// If it doesn't exist, just use the default value.
+		foreach ( $settings->all() as $key => $default ) {
+			$value = get_option( $key, $default );
+			$migrated_settings[ $key ] = $value;
+
+			// Delete the option in case it existed to clean up
+			delete_option( $key );
```

Comment:
You should consider delaying the deletion until you've confirmed that they were saved successfully. If an error occurred on the last setting, all the others would be lost. Only needs to add one line of code:

```php
    $to_delete[] = $key;
}
...
// after options are confirmed saved...
array_map( 'delete_option', $to_delete );
```